arrow/src/json/reader.rs

```diff
@@ -1123,6 +1148,10 @@ impl Decoder {
             .iter()
             .filter(|field| projection.is_empty() || projection.contains(field.name()))
             .map(|field| {
+                let format_string = match field.metadata() {
+                    Some(meta) => meta.get("format_string"),
+                    _ => None,
+                };
```

Comment:
I don't think it matters, but I think you can rewrite this a bit more succinctly like this (untested):

```suggestion
                let format_string = field.metadata()
                  .map(|meta| meta.get("format_string"));
```